frontend/src/slices/auth/actions.ts

```diff
@@ -14,16 +15,20 @@ import {
 import { name as sliceName } from './auth.slice.js';
 
 const signUp = createAsyncThunk<
-  CustomerSignUpResponseDto | BusinessSignUpResponseDto,
+  CustomerSignUpResponseDto | BusinessSignUpResponseDto | undefined,
   {
     payload: CustomerSignUpRequestDto | BusinessSignUpRequestDto;
     mode: ValueOf<typeof AuthMode>;
   },
   AsyncThunkConfig
->(`${sliceName}/sign-up`, ({ payload, mode }, { extra }) => {
-  const { authApi } = extra;
+>(`${sliceName}/sign-up`, async ({ payload, mode }, { extra }) => {
+  const { authApi, notification } = extra;
 
-  return authApi.signUp(payload, mode);
+  try {
+    return await authApi.signUp(payload, mode);
+  } catch (error) {
+    notification.warning(getErrorMessage(error));
+  }
```

Comment:
```suggestion
  } catch (error: unknown) {
    notification.warning(getErrorMessage(error));
    throw error;
  }
```
Action should be rejected if something went wrong. If we won't throw an error - it will be fulfilled